daffodil-core/src/test/scala/org/apache/daffodil/dsom/walker/BasicWalker.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.dsom.walker
+
+import scala.collection.mutable.ArrayBuffer
+
+class BasicWalker(ignoreTypeWrappers: Boolean = false, onlyElements: Boolean = false) extends AbstractDSOMWalker {
+
+  var nodeArr: ArrayBuffer[Either[TermView, TypeView]] = ArrayBuffer()
+
+  override protected def onWalkBegin(root: RootView): Unit = {}
+
+  override protected def onWalkEnd(root: RootView): Unit = {}
+
+  override def onTermBegin(termElement: TermView): Unit = {
+    if (onlyElements) {
+      termElement match {
+        case _: ElementBaseView => nodeArr += Left(termElement)
+        case _ =>
+      }
+    } else {
+      nodeArr += Left(termElement)
+    }
+  }
```

Comment:
I think maybe a cleaner or more scala-y way to write this would be something like this:
```scala
termElement match {
  case _: ElementBaseView => nodeArr += Left(termElement)
  case _=> if (!onlyElements) nodeArr += Left(termElement)
}
```

Reply:
Yes, that is much cleaner - I implemented this along with another case statement based on a new boolean for the tester class in the latest revision.